### DISCUSSION TITLE
Panel authorization based on user roles

### DISCUSSION THREAD

**Discussion by mohandtharsha** [filamentphp/filament#8283] [Help]
I want to access panels based on roles 
for example if admin user logs in using http://127.0.0.1:8000/employer/login it should not login
for example if employer user logs in using http://127.0.0.1:8000/admin/login it should not login
Kindly help me someone How to achieve this in filament version 3?

**Comment by leandrocfe** [comment]
You could add the logic here [authorizing-access-to-the-panel](https://filamentphp.com/docs/3.x/panels/users#authorizing-access-to-the-panel)
or [customize the login class](https://filamentphp.com/docs/3.x/panels/users#customizing-the-authentication-features) like `$panel->login(CustomLogin::class)`

**Reply by mohandtharsha** [reply]
How to do that?

**Reply by leandrocfe** [reply]
Did you read the docs?

**Reply by mohandtharsha** [reply]
I am new to filament can you help me with your code please

**Reply by leandrocfe** [reply]
Sorry, follow the docs. It is simple

**Comment by ghost** [comment]
It looks like you have two different types of roles, each with its own admin panel, and you want users to access the dashboard based on their roles. You can achieve this by conditional checking the `panelId` and the other user requirements to access the dashboard.

`App\Models\User.php`

```php
public function canAccessPanel(Panel $panel): bool
    {
        if ($panel->getId() === 'admin') {
            return str_ends_with($this->email, '@yourdomain.com') && $this->hasVerifiedEmail();
        } else if ($panel->getId() === 'employer') {
            return str_ends_with($this->email, '@yourdomain.com') && $this->hasVerifiedEmail();
        }

        return false; // default case, in case $panel->getId() is neither 'admin' nor 'owner'
    }
```

Before allowing access to the dashboard, it checks the panel ID and the user's access requirements. I hope this is helpful.

( If your using spatie laravel permission package you can check `$this->hasRole(['employer'])` However, you can use your own conditions you prefer and using **spatie permission is not required**. It's up to you. I just provide as an addition info 😁. )
